ttkwidgets/shell.py

```diff
@@ -0,0 +1,205 @@
+import tkinter as tk
+import tkinter.font as tkfont
+from collections import defaultdict
+
+
+class Shell(tk.Canvas):
```

Comment:
Quite some time ago I started implementing a similar widget on the `Console` branch. Personally, I would add quite some features, including pressing up and down to cycle through command history, using the `tab` to have some form of completion...
The `textvariable` is modifiable by the creator of the widget, which inherently allows the creator to implement all these things. It is, however, not easy to do things with `\b` or other special characters. Perhaps it is worth considering to implement these things in the base widget.